File: Java/refactoring_exercise_4/src/main/java/za/co/entelect/refactoring4/domain/ChequeAccount.java
package za.co.entelect.refactoring4.domain;

public class ChequeAccount extends BankAccount {

    public ChequeAccount(Long balanceInCents, double creditInterestsRate, double debitInterestRate, long accountFee) {
        super(balanceInCents, creditInterestsRate, debitInterestRate, accountFee);
    }

    @Override
    public AccountType getAccountType() {
        return AccountType.CHEQUE;
    }
}

File: C#/Refactoring/refactoring_exercise_3/za/co/entelect/refactoring3/domain/ChequeAccount.cs
namespace refactoring_exercise_3.za.co.entelect.refactoring3.domain
{

    public class ChequeAccount : BankAccount
    {

        public ChequeAccount(long balanceInCents, double creditInterestsRate, double debitInterestRate, long accountFee) :
            base(balanceInCents, creditInterestsRate, debitInterestRate, accountFee)
        {
        }

        public override AccountType GetAccountType()
        {
            return AccountType.Cheque;
        }
    }
}